packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/state_var.rs

```diff
@@ -535,17 +559,66 @@ impl<T: Default + Clone> StateVarReadOnlyView<T> {
     }
 }
 
+#[enum_dispatch]
+pub trait QueryUpdateRequests {
+    /// Reset variable that tracks whether or not an update has been requested
+    fn reset_queued_updates(&mut self);
+
+    fn return_indices_with_queued_updates(&self) -> Vec<usize>;
+}
+
+impl<T: Default + Clone> QueryUpdateRequests for StateVarReadOnlyView<T> {
+    /// Reset 'update_has_been_queued` to false
+    fn reset_queued_updates(&mut self) {
+        self.update_has_been_queued = false;
+    }
+
+    fn return_indices_with_queued_updates(&self) -> Vec<usize> {
```

Comment:
Why is this a vec? Will there be times when multiple things are actually returned?

Reply:
Yes, for the case when on vectors, below.